core/Plugin/Archiver.php

```diff
@@ -83,6 +86,73 @@ public function __construct(ArchiveProcessor $processor)
         $this->enabled = true;
     }
 
+    private function getPluginName()
+    {
+        $className = get_class($this);
+        $parts = explode('\\', $className);
+        $parts = array_filter($parts);
+        $plugin = $parts[2];
+
+        if (!Manager::getInstance()->isPluginLoaded($plugin)) {
+            return null;
+        }
+
+        return $plugin;
+    }
+
+    /**
+     * @return ArchiveProcessor\RecordBuilder[]
+     * @throws \DI\DependencyException
+     * @throws \DI\NotFoundException
+     */
+    private function getRecordBuilders()
+    {
+        $transientCache = Cache::getTransientCache();
+        $cacheKey = 'Archiver.RecordBuilders';
+
+        $recordBuilders = $transientCache->fetch($cacheKey);
+        if ($recordBuilders === false) {
+            $recordBuilderClasses = Manager::getInstance()->findMultipleComponents('RecordBuilders', ArchiveProcessor\RecordBuilder::class);
+
+            $recordBuilders = array_map(function ($className) {
+                if ((new \ReflectionClass($className))->getConstructor()->getNumberOfRequiredParameters() == 0) {
+                    return StaticContainer::getContainer()->make($className);
+                }
+            }, $recordBuilderClasses);
+            $recordBuilders = array_filter($recordBuilders);
+
+            /**
+             * Triggered to add new RecordBuilders that cannot be picked up automatically by the platform.
+             * If you define RecordBuilders that take a parameter, for example, an ID to an entity your plugin
+             * manages, use this event to add instances of that RecordBuilder to the global list.
+             *
+             * **Example**
+             *
+             *     public function addRecordBuilders(&$recordBuilders)
+             *     {
+             *         $recordBuilders[] = new MyParameterizedRecordBuilder($idOfThingToArchiveFor);
+             *     }
+             *
+             * @param ArchiveProcessor\RecordBuilder[] $reports An array of dimensions
+             * @api
+             */
+            Piwik::postEvent('Archiver.addRecordBuilders', [&$recordBuilders]);
```

Comment:
not too important but we'd also later need a `filterRecordBuilders` event.

Reply:
:+1: